.github/workflows/merge-commits-to-next-branch.yml

```diff
@@ -0,0 +1,134 @@
+
+jobs:
+  dequeue:
+    runs-on: ubuntu-latest
+    steps:
+      steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - run: npm ci
+      - run: npm install -g @azure/storage-queue
+      - uses: actions/github-script@v6
+        with:
+          script: |
+            const { QueueClient, QueueServiceClient } = require('@azure/storage-queue')
+            const connectionString = `${{ secrets.AZURE_STORAGE_CONNECTION_STRING }}`;
+            const queueName = `${{ secrets.AZURE_QUEUE_NAME }}`;
+            const queueServiceClient = QueueServiceClient.fromConnectionString(connectionString);
+            const queueClient = queueServiceClient.getQueueClient(queueName);
+            var receivedMessages = await queueClient.receiveMessages();
+            const message = receivedMessages.receivedMessageItems[0];
+            $COMMIT_INFO = JSON.parse(message.messageText);
+  create-branch:
+    runs-on: ubuntu-latest
+    if: ( $COMMIT_INFO.label == 'queued' )
+    steps:
+      - name: Create branch
+        uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # pin@v6
+        with:
+          script: |
+            git fetch origin
+            git checkout test_next
+            git checkout -b main-next-$COMMIT_INFO.sha
+            git merge $COMMIT_INFO.sha --no-ff
+  reset-branch:
+    runs-on: ubuntu-latest
+    if: ( $COMMIT_INFO.label == 'queued' && needs.create-branch.result == 'failure'
+      )
+    steps:
+      - name: Reset main to a commit
+        uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # pin@v6
+        with:
+          script: |
+            git fetch origin
+            git checkout main
+            git branch -D main-next-$COMMIT_INFO.sha
+            git checkout -b main-next-$COMMIT_INFO.sha
+            git reset --hard $COMMIT_INFO.sha
+  createPullRequest:
+    runs-on: ubuntu-latest
+    needs: [ create-branch ]
+    if: ( $COMMIT_INFO.label == 'queued' && needs.create-branch.result == 'success'
+      )
+    steps:
+      - name: Create Pull Request
+        uses: actions/github-script@7a5c598405937d486b0331594b5da2b14db670da # pin@v6
+        with:
+          script: |
+            const { repo, owner } = context.repo;
+            const newPr = await github.rest.pulls.create({
+              title: '[Main:Next] Integrate $COMMIT_INFO.sha',
+              'sonalivdeshpande',
+              repo,
+              head: 'main-next-$COMMIT_INFO.sha',
+              base: 'test_next',
+              body: [
+                'This PR is auto-generated by',
+                '[actions/github-script](https://github.com/actions/github-script).'
+              ].join('\n')
+            });
+            github.rest.issues.addLabels({
+              owner,
+              repo,
+              issue_number: newPr.data.number,
+              labels: ['merge-with-next', 'automated pr']
```

Comment:
Since we use spaces elsewhere in our labels, I suggest for consistency we use `merge with next` or `merge: next`.